packages/wallet-sdk/src/components/ConnectDialog/ConnectDialog.tsx

```diff
@@ -0,0 +1,95 @@
+// Copyright (c) 2018-2022 Coinbase, Inc. <https://www.coinbase.com/>
+// Licensed under the Apache License, version 2.0
+
+import clsx from "clsx";
+import { h } from "preact";
+import { useEffect, useState } from "preact/hooks";
+
+import { ConnectContent } from "../ConnectContent/ConnectContent";
+import { TryExtensionContent } from "../TryExtensionContent/TryExtensionContent";
+import css from "./ConnectDialog-css";
+
+type ConnectDialogProps = {
+  darkMode: boolean;
+  version: string;
+  sessionId: string;
+  sessionSecret: string;
+  linkAPIUrl: string;
+  isOpen: boolean;
+  isConnected: boolean;
+  isParentConnection: boolean;
+  chainId: number;
+  connectDisabled: boolean;
+  onCancel: (() => void) | null;
+};
+
+export const ConnectDialog = (props: ConnectDialogProps) => {
+  const { isOpen, darkMode } = props;
+  const [isContainerHidden, setContainerHidden] = useState(!isOpen);
+  const [isDialogHidden, setDialogHidden] = useState(!isOpen);
+
+  useEffect(() => {
+    const timers = [
+      window.setTimeout(() => {
+        setDialogHidden(!isOpen);
+      }, 10),
+    ];
+
+    if (isOpen) {
+      setContainerHidden(false);
+    } else {
+      timers.push(
+        window.setTimeout(() => {
+          setContainerHidden(true);
+        }, 360),
+      );
+    }
+
+    return () => {
+      timers.forEach(window.clearTimeout);
+    };
+  }, [props.isOpen]);
```

Comment:
just curious - what are these timers for again?

Reply:
They're used to control the styles of the modal. They were copied over from the previous dialog